File: sdk_files/v33/javascript/ai_model_credits.js
export class AiModelCredits {
  constructor(ctx) { this._ctx = ctx; }

  async getModelCreditRelationship(modelName) {
    this._ctx.requireTokens();
    const path = `/api/v1/model_credit/${encodeURIComponent(modelName)}`;
    const { data } = await this._ctx.client.request(path, 'GET', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }

  async upsertModelCreditRelationship({ modelName, newCreditValue } = {}) {
    this._ctx.requireTokens();
    let path = `/api/v1/model_credit/upsert`;
    const query = new URLSearchParams();
    if (modelName !== undefined) query.set('model_name', modelName);
    if (newCreditValue !== undefined) query.set('new_credit_value', newCreditValue);
    const qs = query.toString();
    if (qs) path += '?' + qs;
    const { data } = await this._ctx.client.request(path, 'POST', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }
}

File: sdk/v1.0.9/python/mediaviz_sdk/ai_model_credits.py
from __future__ import annotations
from typing import Any
from urllib.parse import quote, urlencode
import httpx

from .errors import handle_response


class AiModelCredits:
    def __init__(self, ctx) -> None:
        self._ctx = ctx

    def get_model_credit_relationship(self, model_name: str) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/model_credit/' + quote(str(model_name), safe='')
        return self._ctx.client.request(path, 'GET', self._ctx.access_token, self._ctx.refresh_token).data

    def upsert_model_credit_relationship(
        self,
        model_name: str | None = None,
        new_credit_value: int | None = None,
    ) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/model_credit/upsert'
        _q: dict[str, str] = {}
        if model_name is not None:
            _q['model_name'] = str(model_name)
        if new_credit_value is not None:
            _q['new_credit_value'] = str(new_credit_value)
        if _q:
            path += '?' + urlencode(_q)
        return self._ctx.client.request(path, 'POST', self._ctx.access_token, self._ctx.refresh_token).data
